src/CommunityToolkit.Maui.UnitTests/Animations/BaseAnimationTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Diagnostics;
+using CommunityToolkit.Maui.Animations;
+using CommunityToolkit.Maui.UnitTests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace CommunityToolkit.Maui.UnitTests.Animations;
+
+public abstract class BaseAnimationTests<TAnimation> : BaseTest where TAnimation : BaseAnimation, new()
+{
+	protected virtual TAnimation CreateAnimation() => new();
+
+	[Fact]
+	public async Task LengthShouldDictateFullAnimationLength()
+	{
+		var animation = CreateAnimation();
+
+		Label label = new();
+
+		label.EnableAnimations();
+
+		var stopwatch = new Stopwatch();
+		stopwatch.Start();
+		await animation.Animate(label);
+		stopwatch.Stop();
+
+		stopwatch.ElapsedMilliseconds.Should().BeCloseTo(animation.Length, 50);
```

Comment:
can we get this value from some constant in Animation from library code?

Reply:
This might highlight that the code doesn't read well. the `Length` property of the `BaseAnimation` is that value we pass into the constructor. The `50` is my attempt at verifying that the completed duration was within 50 milliseconds of the expected duration. Perhaps this could be a calculated percentage instead. I still need to find a reliable way of verifying this because running the tests locally don't yield positive test results for all yet.